### PR TITLE
FIX: find_outliers

### PR DESCRIPTION
**Pull request by dengemann**
cc @fraimondo

### CODE REVIEW

File: mne/preprocessing/bads.py [hunk 1]
```diff
@@ -15,18 +19,18 @@ def find_outliers(X, threshold=3.0):
         The scores for which to find outliers.
     threshold : float
         The value above which a feature is classified as outlier.
+    max_iter : int
+        The maximum number of iterations.
```
Comment (dengemann) [inline mne/preprocessing/bads.py]: needed because some detection algorithms only rely one iteration, e.g. FASTER

File: mne/preprocessing/bads.py [hunk 2]
```diff
@@ -15,18 +19,18 @@ def find_outliers(X, threshold=3.0):
...
     bad_idx : np.ndarray of int, shape (n_features)
         The outlier indices.
     """
-    max_iter = 2
     my_mask = np.zeros(len(X), dtype=np.bool)
-    X = np.abs(X)
     for _ in range(max_iter):
         X = np.ma.masked_array(X, my_mask)
-        this_z = stats.zscore(X)
+        this_z = np.abs(stats.zscore(X))
```
Comment (dengemann) [inline mne/preprocessing/bads.py]: evil ... outliers with lower than normal z-scores have been ignored. relevant for the correlation metric or low variance @wmvanvliet

**Comment by dengemann** [comment]
cc @wmvanvliet @agramfort

**Comment by agramfort** [comment]
looks ok to me but I cannot tell the impact on actual data.

@wmvanvliet we'll need your +1 here

**Comment by dengemann** [comment]
@agramfort the impact is positive, you will detect more true outliers.

2015-01-31 16:30 GMT+01:00 Alexandre Gramfort <notifications@github.com>:

**Comment by wmvanvliet** [comment]
Good catch. Go ahead and commit.

**Pull request merged**